test/setup-beam.test.js

```diff
@@ -14,6 +14,47 @@ const core = require('@actions/core')
 const setupBeam = require('../src/setup-beam')
 const { problemMatcher } = require('../matchers/elixir-matchers.json')
 
+const matrix = {
+  otp: {
+    'ubuntu-18.04': parseBuild('test/otp/ubuntu-18.04/builds.txt'),
+    'ubuntu-20.04': parseBuild('test/otp/ubuntu-20.04/builds.txt'),
+    'ubuntu-22.04': parseBuild('test/otp/ubuntu-22.04/builds.txt'),
+    windows: parseReleases('test/otp/releases.json'),
+  },
+  elixir: parseBuild('test/elixir/builds.txt'),
+  gleam: parseReleases('test/gleam/releases.json'),
+  rebar3: parseReleases('test/rebar3/releases.json'),
+}
+
+function parseBuild(version) {
+  let versions = {}
+  fs.readFileSync(path.join(process.cwd(), version), 'utf8')
+    .trim()
+    .split('\n')
+    .forEach((line) => (versions[line.split(' ')[0]] = line.split(' ')[0]))
+  return versions
+}
+
+function parseReleases(version) {
+  let versions = {}
+  let data = fs.readFileSync(path.join(process.cwd(), version), 'utf8')
+  let json = JSON.parse(data)
+  if (version.includes('otp')) {
+    json
+      .map((x) => x.assets)
+      .flat()
+      .filter((x) => x.name.match(/^otp_win64_.*.exe$/))
+      .forEach((x) => {
+        let v = x.name.match(/^otp_win64_(.*).exe$/)[1]
+        versions[v] = v
+      })
+    return versions
+  } else {
+    json.map((x) => (versions[x.tag_name] = x.tag_name))
+    return versions
+  }
+}
```

Comment:
This could potentially be exported from the existing code base (`setup-beam.js`) but we can leave that as a future refactor.